src/cli/commands/force/lightning/local/__tests__/setup.test.ts

```diff
@@ -45,19 +46,67 @@ describe('Setup Tests', () => {
     test('Checks that Setup is initialized correctly for iOS', async () => {
         const setup = makeSetup(PlatformType.ios);
         await setup.run(true);
-        expect(platformFlagIsValidMock).toHaveBeenCalledWith(PlatformType.ios);
         expect(executeSetupMock).toHaveBeenCalled();
     });
 
     test('Checks that Setup is initialized correctly for Android', async () => {
         const setup = makeSetup(PlatformType.android);
         await setup.run(true);
-        expect(platformFlagIsValidMock).toHaveBeenCalledWith(
-            PlatformType.android
+        expect(executeSetupMock).toHaveBeenCalled();
+    });
+
+    test('Checks that Setup fails for invalid Platform flag', async () => {
+        const setup = makeSetup('someplatform');
+        let err: any;
+
+        try {
+            await setup.run(true);
+        } catch (error) {
+            err = error;
+        }
+
+        expect(err instanceof SfdxError).toBe(true);
+        expect((err as SfdxError).message).toBe(
+            messages.getMessage('error:invalidInputFlagsDescription')
```

Comment:
I just remembered a technique that Wes taught me before. I should actually do this for tests in `internal` repo too. I think you're stashing `error` into `err` in case that `setup.run` might run without throwing an error and calling `expect`s below the try-catch. We can actually call `expect` inside `catch` by specifying how many assertions we are going to make. Then you can get rid of `err`.

```
expect.assertions(2);
try {
    await setup.run(true);
} catch (error) {
    expect(error instanceof SfdxError).toBe(true);
    expect((error as SfdxError).message).toBe(
        messages.getMessage('error:invalidInputFlagsDescription')
    );
}
```

Reply:
This is nice, will change.